### PR TITLE
Ignore resource prior to scan

### PR DESCRIPTION
**Pull request by eliecharra**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #456 
| ❓ Documentation  | false

## Description

Make driftctl does not enumerate ignored resources types from driftignore.

### CODE REVIEW

**Review by lotoussa — APPROVED** [review]
lgtm

**Review by sundowndev — APPROVED** [review]
tested locally, works fine, good job!

**Comment by moadibfr** [comment]
quick question is it intented that it wont ignore resource filtered by id ? If I have no read right on one kms for example and ignore just this one it would still try to scan it.

When I implemented it prior to our refacto. I filtered in ReadResource that way you can filter by type and id.

**Comment by eliecharra** [comment]
> quick question is it intented that it wont ignore resource filtered by id ? If I have no read right on one kms for example and ignore just this one it would still try to scan it.
> 
> When I implemented it prior to our refacto. I filtered in ReadResource that way you can filter by type and id.

Yeah since deep mode is not our priority anymore I decided to put this behavior aside.
We can discuss about this with the team, and if we absolutely want it, we should do this in another dedicated PR IMHO

**Comment by sjourdan** [comment]
Let's take the KMS case with a user without read rights: how can this user explicitly exclude the key?

**Comment by sjourdan** [comment]
TLDR from Zoom call: 

- it's a deep mode only issue, so it's not prioritized too high
- the access denied resources cases will be handled by #666 so there's no real blocker here

File: pkg/remote/scanner.go [hunk 1]
```diff
@@ -57,6 +60,12 @@ loop:
 
 func (s *Scanner) scan() ([]resource.Resource, error) {
 	for _, enumerator := range s.remoteLibrary.Enumerators() {
+		if s.filter.IsTypeIgnored(enumerator.SupportedType()) {
```
Comment (wbeuil) [inline pkg/remote/scanner.go:63]: Could `s.filter` be nil ?

Comment (eliecharra) [reply pkg/remote/scanner.go:63]: No, we pass a driftignore struct every time, if the driftignore is empty or does not exist we still pass a driftignore struct in `scan.go` but it will have empty matcher. So this should work

**Review by wbeuil — APPROVED** [review]
Tested it and i worked like a charm <3
